Makefile

```diff
@@ -0,0 +1,39 @@
+# Makefile for running typical developer workflow actions.
+# To run actions in a subdirectory of the repo:
+#   make lint build dir=translate/snippets
+
+INTERFACE_ACTIONS="build test lint"
+
+.ONESHELL: #ease subdirectory work by using the same subshell for all commands
+.-PHONY: *
+
+# Default to current dir if not specified.
+dir ?= $(shell pwd)
+
+export GOOGLE_CLOUD_PROJECT = ${GOOGLE_SAMPLES_PROJECT}
+
+build:
+	cd ${dir}
+	mvn compile
+
+test: check-env build
+	cd ${dir}
+	mvn --quiet --batch-mode --fail-at-end clean verify \
+    -Dfile.encoding="UTF-8" \
+    -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+    -Dmaven.test.redirectTestOutputToFile=true \
+    -Dbigtable.projectID="${GOOGLE_CLOUD_PROJECT}" \
+    -Dbigtable.instanceID=instance
+
+lint:
+	cd ${dir}
+	mvn -P lint checkstyle:check
+
+check-env:
+ifndef GOOGLE_SAMPLES_PROJECT
```

Comment:
Currently, the tests in the repo requires another env variable "GOOGLE_APPLICATION_CREDENTIALS" pointing to the Service account JSON key for testing.

Reply:
This may be true for using a Service Account, but they can also be run without this variable, using Application Default Credentials.
I looked for a way to verify ADCs are present, but could not find a way that would catch both cases 😞 
I think we'll have to rely on the [existing setup docs](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/main#set-up) for this aspect.

Reply:
If that's the case, can we add a comment saying the user needs to be authenticated before running the Makefile?

Reply:
Done. I've added a note about this to the top of the file, near the other usage information. I'll also be sure to include it in the docs, which are the next step.